src/utils/dom.ts

```diff
@@ -118,3 +118,16 @@ export function getDescribedByElement<T extends Element>(element: Element): T |
 export function hasLabel(labelEl: HTMLCalciteLabelElement, el: HTMLElement): boolean {
   return labelEl.contains(el);
 }
+
+// set a default icon from a defined set or allow an override with an icon name string
+export function setRequestedIcon(
+  iconObject: Record<string, string>,
+  iconValue: string | boolean,
+  matchedValue: string
+): string {
+  if (iconValue && typeof iconValue === "string" && iconValue !== "") {
```

Comment:
you may be able to skip the first one here. `typeof null` is `"null"` and `typeof undefined` is `"undefined"` so neither would break or be `"string"`, I think.